includes/class-indieauth-client-discovery.php

```diff
@@ -74,45 +75,63 @@ private function parse( $url ) {
 			return $response;
 		}
 
-		$content = wp_remote_retrieve_body( $response );
-
-		if ( class_exists( 'Masterminds\\HTML5' ) ) {
-			$domdocument = new \Masterminds\HTML5( array( 'disable_html_ns' => true ) );
-			$domdocument = $domdocument->loadHTML( $content );
-		} else {
-			$domdocument = new DOMDocument();
-			libxml_use_internal_errors( true );
-			if ( function_exists( 'mb_convert_encoding' ) ) {
-				$content = mb_convert_encoding( $content, 'HTML-ENTITIES', mb_detect_encoding( $content ) );
+		$content_type = wp_remote_retrieve_header( $response, 'content-type' );
+		if ( 'application/json' === $content_type ) {
+			$json = json_decode( wp_remote_retrieve_body( $response ), true );
+			if ( ! is_array( $json ) && empty( $json ) ) {
```

Comment:
Guessing maybe this should be `|| empty( $json )` ? Like, you wanna return when it's either empty _or_ not an array (in other words, it has to be both an array and non-empty).